lte/gateway/python/magma/pipelined/ng_manager/node_state_manager.py

```diff
@@ -0,0 +1,179 @@
+"""
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import logging
+import netifaces
+from typing import NamedTuple, Dict
+
+from lte.protos.session_manager_pb2 import (
+    UPFNodeState,
+    UPFAssociationState,
+    UPFFeatureSet,
+    UserPlaneIPResourceSchema)
+
+from google.protobuf.timestamp_pb2 import Timestamp
+from magma.pipelined.set_interface_client import send_node_state_association_request
+from ryu.lib import hub
+
+EXP_BASE = 3
+
+class NodeStateManager:
+    """
+    This controller manages node state information
+    and reports to SMF.
+    """
+
+    TEID_RANGE_INDICATION = 0
+    TEID_RANGE_VALUE = 0
+    ASSOC_MAX_RETRIES = 40
+
+    GetLocalConfig = NamedTuple(
+        'GetLocalConfig',
+        [('downlink_ip', str), ('node_identifier', str)])
+
+    def __init__(self, loop, sessiond_setinterface, config):
+        self._loop = loop
```

Comment:
for init can you group public vars init first and all the private ones(with _) after that

Reply:
Comments addressed.